perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -3240,14 +3259,17 @@ public void updateApplicationUser(PerunSession sess, Application app) {
 
 	public void updateFormItemData(PerunSession sess, int appId, ApplicationFormItemData data) throws RegistrarException, PrivilegeException {
 
-		//Authorization
-		if (!AuthzResolver.authorizedInternal(sess, "updateFormItemData_int_ApplicationFormItemData_policy")) {
-			throw new PrivilegeException(sess, "updateFormItemData");
-		}
 
 		Application app = getApplicationById(sess, appId);
 		if (AppState.APPROVED.equals(app.getState()) || AppState.REJECTED.equals(app.getState())) throw new RegistrarException("Form items of once approved or rejected applications can't be modified.");
 
+		//Authorization
+		if (app.getGroup() == null && (!AuthzResolver.authorizedInternal(sess, "updateFormItemData_int_ApplicationFormItemData_policy", app.getVo()))) {
```

Comment:
group==null and is authorized => will go to `else if ` branch and i guess would fail on the null group